### PR TITLE
BugFix: adjoint metric tensor with jax

### PR DESCRIPTION
**Pull request by astralcai**
**Context:**
The `adjoint_metric_tensor` transform does not work with jax variables because jax variables are not considered trainable parameters until they become tracers.

**Description of the Change:**
1. Add a custom expand transform to `adjoint_metric_tensor` that expands trainable parameters based on argnums.
2. Add an optional `use_argnum_in_expand` argument to transform programs to determine whether or not to perform `jax_argnums_to_tape_trainable` on the parameters.

**Benefits:**
BugFix

**Possible Drawbacks:**
Adding yet another keyword argument to `transform` may make code look messy.

**Related GitHub Issues:**
https://github.com/PennyLaneAI/pennylane/issues/5197

**Related Shortcut Stories:**
[sc-56734]

### CODE REVIEW

File: pennylane/gradients/adjoint_metric_tensor.py [hunk 1]
```diff
@@ -54,13 +60,28 @@ def _group_operations(tape):
...
 @partial(
     transform,
+    expand_transform=_expand_trainable_multipar,
     classical_cotransform=_contract_metric_tensor_with_cjac,
     is_informative=True,
+    use_argnum_in_expand=True,
 )
 def adjoint_metric_tensor(
-    tape: qml.tape.QuantumTape,
+    tape: qml.tape.QuantumTape, argnums=None
```
Comment (rmoyard) [inline pennylane/gradients/adjoint_metric_tensor.py:84]: I don't think it is necessary here, because the qnode will use argnums expand them and set them on the tape directly. In group operation you can keep the old logic for everything that is not jax and set argnums=tape.trainable_params for Jax.

**Review by rmoyard — CHANGES_REQUESTED** [review]
It looks good! I don't think it is necessary to argnums as arguments though

File: pennylane/gradients/adjoint_metric_tensor.py (hunk 1 above)
Comment (astralcai) [reply pennylane/gradients/adjoint_metric_tensor.py:84]: See `test_adjoint_metric_tensor.py` for `test_correct_output_qnode_jax` with parameters `fubini_ansatz4` and its corresponding params `(np.array(-0.1735, requires_grad=True), np.array([-0.1735, -0.2846, -0.2846], requires_grad=True))`. In this test case, two parameters are passed to the qnode, and by default, only the first parameter is considered trainable, and this is due to `argnums = [0] if qnode.interface in ["jax", "jax-jit"] and argnums is None else argnums` this line in `transform_program.py`. So if there are more than one jax arrays that are all supposed to be trainable, we need to provide users with the option of specifying `argnums`.

Comment (rmoyard) [reply pennylane/gradients/adjoint_metric_tensor.py:84]: Look at the other transforms, you can specify `argnums` even when they are not argument of the function. Because apply the transforms just populates the transform program. The qnode uses the argnums but not the transform itself.

Comment (astralcai) [reply pennylane/gradients/adjoint_metric_tensor.py]: Thanks! I just made the update.

**Review by rmoyard — APPROVED** [review]
Thanks @astralcai , it looks good to me :100:

**Review by timmysilv — APPROVED** [review]
looks good!
